### PR TITLE
Update conformance testsuite to latest Keycloak, version to 4.1.12 and support for online conformance testsuite

### PR DESCRIPTION
**Pull request by mposolda**
* Update realm JSON to pass with latest Keycloak master
 
* Update Keycloak dockerfile ciphers to match with FAPI 1 Advanced Final - section 8.5

* Update conformance testsuite to 4.1.12

* Add support and instructions for running against production conformance testsuite from www.certification.openid.net

### CODE REVIEW

**Comment by mposolda** [comment]
@tnorimat Do you please have a chance to review this?

It is still draft as it should not be merged before Keycloak 14 is released. I've closed https://github.com/keycloak/kc-sig-fapi/pull/175 and replaced with this one.

File: .env [hunk 1]
```diff
@@ -1,5 +1,5 @@
 OPENID_GIT_URL=https://gitlab.com/openid/conformance-suite.git
-OPENID_GIT_TAG=${OPENID_GIT_TAG:-release-v4.1.11}
+OPENID_GIT_TAG=${OPENID_GIT_TAG:-release-v4.1.12}
```
Comment (VinodAnandan) [inline .env:2]: IMO, we should use the latest (4.1.13) version available - https://gitlab.com/openid/conformance-suite/-/tags/release-v4.1.13

Comment (mposolda) [reply .env:2]: @VinodAnandan Good point! I will update the version once I check this setup is working with 4.1.13. I will probably do it aftern Keycloak 14 release, so I am able to try everything together before this is merged (Keycloak 14 release together with 4.1.13 testsuite)

**Review by tnorimat — COMMENTED** [review]
@mposolda I've added some review comments. Could you check them?

File: docker-compose.yml [hunk 2]
```diff
@@ -34,8 +35,8 @@ services:
      - ./https/server.pem:/etc/x509/https/tls.crt
      - ./https/server-key.pem:/etc/x509/https/tls.key
      - ./https/client-ca.pem:/etc/x509/https/client-ca.crt
-#     - ../bin/keycloak-conformance-test:/opt/jboss/keycloak
-    command: "-b 0.0.0.0 -Djboss.socket.binding.port-offset=1000 --debug -Dkeycloak.profile=preview"
+     - /tmp/keycloak-14.0.0-SNAPSHOT:/opt/jboss/keycloak
+    command: "-b 0.0.0.0 -Djboss.socket.binding.port-offset=1000 --debug *:8787 -Dkeycloak.profile=preview -Dkeycloak.frontendUrl=${KEYCLOAK_FRONTEND_URL}"
```
[the remark this answers is not in the document]
Comment (mposolda) [reply docker-compose.yml:39]: That is good point. I was thinking exactly about this. The only reason, why I kept `-Dkeycloak.profile=preview` is, that this conformance tests project can be potentially used in the future to run other kind of tests like CIBA, PAR and others, which might rely on the preview features.

So the option is to either:
- Keep the `-Dkeycloak.profile=preview` in this PR to make sure that in the future, the conformance testsuite will work smoothly for the scenarios related to preview features (CIBA, PAR etc).
- Remove `-Dkeycloak.profile=preview` in this PR and then potentially add it later in some other PR when we want to run the testsuite against stuff like CIBA or PAR.

WDYT? Let me know what is your preference and I will go with it in this PR.

[the remark this answers is not in the document]
Comment (tnorimat) [reply docker-compose.yml:39]: IMO, Keeping the `-Dkeycloak.profile=preview` in this PR is better for considering the future.

File: .env (hunk 1 above)
Comment (mposolda) [reply .env]: @VinodAnandan @tnorimat Updated to 4.1.13. Also updated Keycloak to 14 in this PR. Thanks!

File: README.md [hunk 3]
```diff
@@ -125,15 +129,76 @@ To access to Keycloak and Resource server with FQDN, modify your `hosts` file in
...
-9. Click `Create Test Plan` button and follow the instructions. To proceed with the tests, You can authenticate using `john` account with password `john`. When rejecting authentication scenario, you can use `mike` account with password `mike`. In this case, you need to click `No` button to cancel the authentication in the consent screen.
-
+9. Click `Create Test Plan` button and follow the instructions.
+10. If you used `FAPI1-Advanced-Final`, it is recommended to use Keycloak global builtin client profile `fapi-1-advanced` instead of the
+`fapi-1-rw-ID2`, which is suitable for the `FAPI-RW-ID2` test profile. To do this, you can login to Keycloak admin console (should be available
+usually under `https://as.keycloak-fapi.org/auth`) as admin/admin. Then go to Realm  `Test` -> `Client Policies` -> `Policies` -> `fapi-1-0-policy`. Then delete `fapi-1-rw-ID2` client profile
+from this policy and add `fapi-1-advanced` instead of it. 
+11. To proceed with the tests, You can authenticate using `john` account with password `john`.
```
[the remark this answers is not in the document]
Comment (mposolda) [reply README.md]: Updated, Thanks!

File: docker-compose.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mposolda) [reply docker-compose.yml:39]: @tnorimat Thanks Takashi. I am resolving this conversation and keeping `-Dkeycloak.profile=preview` in this PR

**Comment by VinodAnandan** [comment]
LGTM

**Comment by mposolda** [comment]
@tnorimat @VinodAnandan I've increased the version of Keycloak to 14 and the version of the testsuite to 4.1.13. I can confirm that conformance testsuite is passing with this configuration and it passed for the Takashi as well.

I am unmarking this PR from the draft stage, it is ready for merge from my point of view unless you have additional concerns?

**Review by tnorimat — APPROVED** [review]
LGTM

**Comment by tnorimat** [comment]
@mposolda Thank you!

**Comment by mposolda** [comment]
@tnorimat @VinodAnandan Thanks for the feedback and review of the PR!

**Pull request merged**